prowler/providers/azure/services/iam/iam_custom_role_permits_administering_resource_locks/iam_custom_role_permits_administering_resource_locks.py

```diff
@@ -0,0 +1,31 @@
+from re import search
+
+from prowler.lib.check.models import Check, Check_Report_Azure
+from prowler.providers.azure.services.iam.iam_client import iam_client
+
+
+class iam_custom_role_permits_administering_resource_locks(Check):
+    def execute(self) -> Check_Report_Azure:
+        findings = []
+        for subscription, roles in iam_client.roles.items():
+            roles = [role for role in roles if role.type == "CustomRole"]
+            for role in roles:
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.resource_id = role.id
+                report.resource_name = role.name
+                for scope in role.assignable_scopes:
+                    if search("^/.*", scope):
+                        for permission_item in role.permissions:
+                            for action in permission_item.actions:
```

Comment:
Out of curiosity, do we need to loop for all roles, then again for roles, scopes, permissions and actions? Maybe we can build something simpler in the service and easier to get that data. Maybe not, just thinking out loud.